src/main.rs

```diff
@@ -58,13 +58,38 @@ where
 {
     let mut args = vec!["check".to_owned()];
 
+    let mut fix = false;
+    let mut unstable_options = false;
+
     for arg in old_args.by_ref() {
-        if arg == "--" {
-            break;
+        match arg.as_str() {
+            "--fix" => {
+                fix = true;
+                continue;
+            },
+            "--" => break,
+            // Cover -Zunstable-options and -Z unstable-options
+            s if s.ends_with("unstable-options") => unstable_options = true,
+            _ => {},
         }
+
         args.push(arg);
     }
 
+    if fix {
+        if unstable_options {
+            args[0] = "fix".to_owned();
+        } else {
+            panic!("Usage of `--fix` requires `-Z unstable-options`");
+        }
+    }
```

Comment:
The argument building from the start of the function until here could be extracted into a separate function and unit tested maybe?

Reply:
will do